### ISSUE TITLE
Support for local RAG pipelines

### ISSUE THREAD

**Issue by manojkr19** [ollama-webui/ollama-webui#138]
I've several local RAG pipelines with different index, retrieval strategies and tested against specific model.  Would like to point this UI to those pipelines via custom APIs (similar to Ollama api with reverse proxy).  An option of API key for authentication would be great.

**Comment by tjbck** [comment]
Hi, Thanks for the suggestion but I'm not exactly sure what you meant. Could you rephrase your wording and provide us with example use cases? Thanks!

**Comment by manojkr19** [comment]
we have APIs that takes user prompt, retrieves local document and creates the prompt with local knowledge and sends request to Open AI or local models etc.  Basically tons of prompt engineering wrapped around the model.  Would like this UI to point to those APIs as a chat interface.  Does this help.  You can assume it's very similar to calling the Ollama API with the user chat history and retrieving the response and showing it to the user, but instead of directly talking to the model API, the UI would talk to a custom API.

**Comment by tjbck** [comment]
I'm actively working on a built-in RAG feature for the webui, once that's completed I'll try to implement your request feature if it aligns with the webui, Let's continue our discussion here: #31. Thanks!
